### ISSUE TITLE
docs: clarify what is meant by `data` in [CLI](https://docs.skuid.com/latest/en/skuid/cli/#skuid-cli) docs

### ISSUE THREAD

**Issue by techfg** [skuid/skuid-cli#201]
In the introduction of the CLI in the [CLI](https://docs.skuid.com/latest/en/skuid/cli/#skuid-cli) docs, it states that the CLI can we used for `...retrieving and deploying Skuid objects (data and metadata) on Skuid NLX sites`.

What isn't clear, from the above, is what is meant by `data`.  

When creating a Skuid DB, it does not appear that a `retrieve` pulls down the data portion of the database and it doesn't appear that `deploy` can deploy the `data` portion.  For other datasources, it would not make much sense to `retrieve`/`deploy` a data portion of the data source since it would be an external system.

Beyond `metadata` definition files (e.g., `pages/my_page.json, apps/my_app.json, etc.), the only other types of files that appear to be retrieveable/deployable are `pages/*.xml` and metadata types that have "files" (e.g., files/my_javascript.js, site/favicon/my_favicon.ico).

In short, it's unclear what `data` means in the docs so would be helpful to understand:

1. What is meant by `data` in the docs, specifically what type of `data` is supported by the CLI or does the CLI only support `metadata`?
2. Assuming `Skuid DB` data is NOT supported (hopefully it is), what options are there to automate the deployment of the data if the CLI does not support it?
